fix(api): use StateUpdate when mapping legacy module loads

jbleon95 opened the pull request:
# Overview

In RESC-513, a customer reported a magnetic module not appearing on the deck layout or in LPC after updating from robot release 8.5.0 to 8.7.0.

Upon further investigation, the problem was found to be related to the refactor of updating PE state from the command results to `StateUpdate` handling. The change was made in #18639 but the corresponding updates to the legacy command mapper were not made. This has the effect of breaking anything that relies on that module field of the analysis results to be filled, which most obviously are the deck layout and LPC.

The fix for this is putting in the legacy command mapper the correct state update, which was already being done for pipette and labware commands.

## Test Plan and Hands on Testing

The customer's protocol in the linked ticket was tested, but this minimal reproducible protocol was tested too and now works as expected:

```python
requirements = {
    "robotType": "OT-2",
    "apiLevel": "2.8",
}

metadata = {
    "protocolName":'OT-2 module loading test',
}

def run(protocol_context):
    tiprack_1 = protocol_context.load_labware('opentrons_96_tiprack_300ul', 8)
    tiprack_2 = protocol_context.load_labware('opentrons_96_tiprack_300ul', 9)

    pipette = protocol_context.load_instrument('p300_multi_gen2', 'left', tip_racks=[tiprack_1, tiprack_2])

    magdeck = protocol_context.load_module('magnetic module gen2', 7)
    mag_plate = magdeck.load_labware('armadillo_96_wellplate_200ul_pcr_full_skirt')
    heater_shaker = protocol_context.load_module('temperatureModuleV2', 1)

    pipette.pick_up_tip()
    pipette.aspirate(50, mag_plate.wells()[0].top(-1))
    pipette.dispense()
    pipette.drop_tip()
```

## Changelog

- call `set_load_module` on `StateUpdate` in legacy command mapper `_map_module_load`

## Review requests


## Risk assessment

Low, this only affects OT-2 protocols on versions 2.13 and below and this fix follows the existing pattern we have for pipette and labware loads, which do work on older versions.

jbleon95 committed:
correctly fill in StateUpdate when mapping legacy module loads

sfoster1 reviewed (approved):
That looks good to me. Are there any remaining command that get mapped without a state update for whatever reason?

mjhuff reviewed (approved):
Ahhh, yeah, this makes sense. Sorry for the hassle, and thank you for fixing!

SyntaxColoring commented on api/src/opentrons/protocol_runner/legacy_command_mapper.py:
```diff
@@ -841,8 +841,16 @@ def _map_module_load(
+            module_id=module_id,
+            definition=loaded_definition,
+            slot_name=module_load_info.deck_slot,
+            requested_model=loaded_model,
```
I think this ought to be `requested_model=requested_model`. So if the user, like, loads a `temperatureModuleV1` but what's actually connected is a `temperatureModuleV2`, we need to pass *both* along. (I think `temperatureModuleV2`, the loaded module, is already implicit in `definition`, so that's already taken care of.)

This matters for Labware Position Check. The logic to decide which offsets to apply works something like "apply this offset to this labware definition if it's on a module that was requested as `temperatureModuleV1`," and that unfortunately means *exactly* `temperatureModuleV1`. I think there's a note about this somewhere in the HTTP API docs.

jbleon95 replied:
Oooh good catch!

SyntaxColoring reviewed (changes requested):
Thank you! One note about `requested_model`. Otherwise LGTM, assuming the analysis snapshot tests get healed to add the missing entries back into `modules`.

jbleon95 committed:
requested model should be the requested model

github-actions[bot] committed:
fix(analyses-snapshot-testing): heal fix_legacy_api_module_mapping snapshots (#20215)

This PR was requested on the PR
https://github.com/Opentrons/opentrons/pull/20210

Co-authored-by: jbleon95 <12025228+jbleon95@users.noreply.github.com>
